cmd/jujud/main.go

```diff
@@ -92,12 +100,62 @@ func getwd() (string, error) {
 	return abs, nil
 }
 
+func getSocket() (sockets.Socket, error) {
+	var err error
+	socket := sockets.Socket{}
+	socket.Network, err = getenv("JUJU_AGENT_SOCKET_NETWORK")
+	if err != nil {
+		return sockets.Socket{}, errors.Annotate(err, "failed to get JUJU_AGENT_SOCKET_NETWORK")
+	}
+	socket.Address, err = getenv("JUJU_AGENT_SOCKET_ADDRESS")
+	if err != nil {
+		return sockets.Socket{}, errors.Annotate(err, "failed to get JUJU_AGENT_SOCKET_ADDRESS")
+	}
+
+	// If we are not connecting over tcp, ignore operator file.
+	if socket.Network != "tcp" {
+		return socket, nil
+	}
+
+	unitName, err := getenv("JUJU_UNIT_NAME")
+	if err != nil {
+		return sockets.Socket{}, err
+	}
+	unit := names.NewUnitTag(unitName)
+
+	baseDir := agent.Dir(cmdutil.DataDir, unit)
+	infoFilePath := filepath.Join(baseDir, provider.OperatorInfoFile)
```

Comment:
I suggest the operator file is just for juju-run where there's no env set up.
For running hook commands, we set up a context via env vars and we could include the cert there. JUJU_AGENT_CERT or something.